pkg/kubelet/eviction/types.go

```diff
@@ -153,18 +154,18 @@ type CgroupNotifier interface {
 type NotifierFactory interface {
 	// NewCgroupNotifier creates a CgroupNotifier that creates events when the threshold
 	// on the attribute in the cgroup specified by the path is crossed.
-	NewCgroupNotifier(path, attribute string, threshold int64) (CgroupNotifier, error)
+	NewCgroupNotifier(logger klog.Logger, path, attribute string, threshold int64) (CgroupNotifier, error)
 }
 
 // ThresholdNotifier manages CgroupNotifiers based on memory eviction thresholds, and performs a function
 // when memory eviction thresholds are crossed
 type ThresholdNotifier interface {
 	// Start calls the notifier function when the CgroupNotifier notifies the ThresholdNotifier that an event occurred
-	Start()
+	Start(ctx context.Context)
 	// UpdateThreshold updates the memory cgroup threshold based on the metrics provided.
 	// Calling UpdateThreshold with recent metrics allows the ThresholdNotifier to trigger at the
 	// eviction threshold more accurately
-	UpdateThreshold(summary *statsapi.Summary) error
+	UpdateThreshold(logger klog.Logger, summary *statsapi.Summary) error
```

Comment:
why some methods passes around context when others - logger. Would it be more consistent to pass context everywhere so when it is needed we can use it in future?

Reply:
At the moment, it doesn’t look like UpdateThreshold needs to use the context anywhere inside. I’m okay with passing the context around.

Reply:
@bart0sh what is the guidance here? I don't feel like passing logger around is a good pattern, but maybe it is used in codebase already?

Reply:
So far we're passing context to public methods and context or logger to the local ones, usually logger is passed when the function is small and only needs a logger.
@pohly suggested today that if function takes a context as a parameter it's expected to support context cancellation. From this perspective, passing a logger if we only need to support contextual logging is preferable.